lib/sensors/IMU.h

```diff
@@ -0,0 +1,207 @@
+/* InvenSense ICM-20948 IMU Sensor Library
+ *
+ * This library provides access to scaled IMU measurement data over SPI.
+ *
+ * TODO:
+ * - calibration
+ * - fixed update frequency?
+ * - DMP support to return orientation quaternion
+ *
+ * Note: This implementation only supports the SPI interface
+ *
+ * Ported from Sparkfun's Arduino library: https://github.com/sparkfun/SparkFun_ICM-20948_ArduinoLibrary
+ * Datasheet: https://invensense.tdk.com/wp-content/uploads/2016/06/DS-000189-ICM-20948-v1.3.pdf
+ *
+ */
+
+#ifndef _IMU_H_
+#define _IMU_H_
+
+#include <stdint.h>
+
+#include <array>
+
+#include "../../config/mbed_config.h"  // so vscode doesn't complain about the SPI class being undefined
+#include "IMU_util.h"
+#include "mbed.h"
+
+#define MAX_SPI_FREQ         7000000
+#define PI                   3.14159265359
+#define ACCELERATION_GRAVITY 9.81
+
+#define ACCEL_SENSITIVITY_FACTOR_0 16384
+#define ACCEL_SENSITIVITY_FACTOR_1 8192
+#define ACCEL_SENSITIVITY_FACTOR_2 4096
+#define ACCEL_SENSITIVITY_FACTOR_3 2048
+#define GYRO_SENSITIVITY_FACTOR_0  131.0
+#define GYRO_SENSITIVITY_FACTOR_1  65.5
+#define GYRO_SENSITIVITY_FACTOR_2  32.8
+#define GYRO_SENSITIVITY_FACTOR_3  16.4
+#define MAG_SENSITIVITY_FACTOR     0.15
+
+// Status codes
+typedef enum {
```

Comment:
Can you see if you can use the MBED error status types here instead? See list at bottom of page https://os.mbed.com/docs/mbed-os/v5.15/apis/error-handling.html#error-code-and-error-status-ranges

Reply:
If it doesn't fit well thats alright, just use MBED statuses in the layer above this

Reply:
:+1: Made a function to convert from custom IMU status codes to mbed error codes.